### ISSUE TITLE
Users often accidentally ignore other users, and it's not very clear in the UI when you've done so

### ISSUE THREAD

**Issue by Half-Shot** [vector-im/riot-web#14633]
There are  two bugs here, one more clearly defined than the other. Often we find users appearing in #synapse-admins:matrix.org who are trying to debug why they can't contact another homeserver, only to find out that they have accidentally ignored a user. It's not clear how they are achieving this, although I can say that the button for ignoring people doesn't give any warnings or feedback so I can see how a misclick can cause this.

Secondly, it's very difficult to see if you've ignored someone in the UI at all. The ignore button changes text to "Unignore", but is styled in the same way as the other buttons. Ideally it would be highlighted *very clearly* when a user is is ignored.

**Comment by ara4n** [comment]
pressing buttons at random on an unlocked phone in your pocket seems to be the most popular way to accidentally ignore people.

**Comment by niquewoodhouse** [comment]
Is this, in the `mx_UserInfo`, the only way people can ignore a user? 

<img width="295" alt="Screenshot 2020-08-03 at 20 09 03" src="https://user-images.githubusercontent.com/1587166/89218156-749de580-d5c5-11ea-994e-d8eec3079d44.png">

When a user is ignored - do they disappear from timelines in the future?  I just tested ignoring someone and I can still search for them in my room list and I can still see them in the `mx_MemberList`.  So I _think_ I can still contact them?

@Half-Shot when people are trying to debug this, are they searching for someone like that and not seeing the user in their lists?  Would be super useful to have the flow documented I think (for me at least).

**Comment by t3chguy** [comment]
Ignoring does hide them from timeline. It does not hide rooms which happened to be named after them such as DMs. They can still see your messages. There is also an /ignore command.

**Comment by aaronraimist** [comment]
Been seeing lots of reports of this lately. This would be another thing that would be improved by https://github.com/matrix-org/matrix-doc/issues/1803 which suggests that ignores should not completely prevent new messages from being sent to you by the server, but instead are filtered out client side and can be displayed (optionally) more like this:

<img width="544" src="https://user-images.githubusercontent.com/5855073/89667627-14978f80-d8a2-11ea-9fac-1111b90f5ac6.png">

rather than being completely missing.

**Comment by Half-Shot** [comment]
This is what Discord does, and people are pretty mixed about it over there. Some people don't want to see messages from users at all if they have ignored them (although the feature could be a setting).

@niquewoodhouse They generally don't look in the user list, because usually the user believes it to be a networking problem rather than the user not actually being present in the room. The user list only tells you that someone is in your room, but it doesn't really tell you if everything is working.

Ideally people would notice that the member panel has a "Unignore" button, but it's not very visible amongst the other buttons.

**Comment by aaronraimist** [comment]
Right. I think that should be an option though, and probably the default when you ignore someone.

A short term relatively easy fix is to add a confirmation step when you click/tap Ignore. (It seems like a lot of these reports are Element Android users so it may be happening there)

**Comment by t3chguy** [comment]
That is how improved ignores in labs work fwiw

**Comment by Half-Shot** [comment]
>A short term relatively easy fix is to add a confirmation step when you click/tap Ignore. (It seems like a lot of these reports are Element Android users so it may be happening there)

I just tried it, turns out it does ask you to confirm it.

**Comment by Half-Shot** [comment]
@t3chguy A possible solution to this bug might be to bring improved ignores out of labs :)

**Comment by aaronraimist** [comment]
@Half-Shot on Android?

On Web 1.7.3 it doesn't for me.

**Comment by Half-Shot** [comment]
Ah on Android yes

**Comment by MazeChaZer** [comment]
>  ignores should not completely prevent new messages from being sent to you by the server

But that isn't the case currently, the filtering is already happening in the client, isn't it? After I unblocked the person I accidentally ignored and cleared my client cache, past messages that were not displayed before came back.

**Comment by t3chguy** [comment]
> the filtering is already happening in the client, isn't it?

Nope it gets applied to future syncs by the server.

**Comment by Biep** [comment]
If the room member list used the same logic as the room list, there could be a category "Ignored" - a kind of analogue to "Low priority' or "History".

**Comment by ivanholmes** [comment]
This bit me today as well, I have no recollection of 'ignoring' the contact and wasted a couple of hours trying to find out what was wrong with the server.

Perhaps it would be useful to remind you that you've ignored someone when you try to message them, or show some kind of indication when you click on the room?

**Comment by aaronraimist** [comment]
@Half-Shot just wanted to follow up on the Android issue, I've created https://github.com/vector-im/element-android/issues/1890 but maybe you can push that forward internally. It looks like the "Ignore" button in Element Android requires a confirmation but if you press and hold on a message, there is a "Block user" button which does the same thing however there is no confirmation required or even any change in the text of the button to indicate that you blocked them.

**Comment by resynth1943** [comment]
Hello! Just chiming in, would like to amend my thoughts.

> This is what Discord does, and people are pretty mixed about it over there. Some people don't want to see messages from users at all if they have ignored them

Yeah... if you do that, you should *really* consider adding a switch. There might be some form of bullying, and I doubt the victim wants to be constantly reminded when the bully is sharing his thoughts ("what did he say?"). That's one of my main gripes with Discord.

> This bit me today as well, I have no recollection of 'ignoring' the contact and wasted a couple of hours trying to find out what was wrong with the server.

Same here. Apparently, I have ignored a user without knowing about it. No idea what's causing it, so I can't really debug it.

**Comment by erikjohnston** [comment]
In Synapse land we see a steady trickle of people reporting bugs due to users accidentally ignoring people. Could I ask that implementing a confirmation dialog or something gets added to the road map pretty please? :angel:

**Comment by robintown** [comment]
From personal experience I know that it's far too easy to accidentally hit the "Reject and ignore" button on an invite instead of just "Reject" and never notice. I wouldn't be surprised if that's how most accidental ignores happen, which is why I would suggest hiding the option under a `...` menu, like in the final design from https://github.com/vector-im/element-web/issues/16869#issuecomment-815032159.

**Comment by MazeChaZer** [comment]
I believe for me it was the ignore button in `mx_UserInfo`, as @niquewoodhouse showed before. Something like this definitely needs a confirmation dialog, ignoring people accidentally is super stressful to debug. I am an experienced user and I really had trouble noticing that messages were missing and then finding the cause. Because I never used the ignore feature before I didn't think of it at all.
